src/elfpy/simulators.py

```diff
@@ -322,23 +326,56 @@ def run_simulation(self, override_dict=None):
                             time_remaining, self.market.time_stretch_constant
                         )
                         action_result = self.market.swap(user_action)
+                        if self.verbose:
+                            print(
+                                f"t={bcolors.HEADER}{self.market.time}{bcolors.ENDC}"+
+                                f" reserves=[x={bcolors.OKBLUE}{self.market.share_reserves}{bcolors.ENDC}"+
+                                f",y={bcolors.OKBLUE}{self.market.bond_reserves}{bcolors.ENDC}]\n"+
+                                f" action: {user_action}\n result: {action_result}"
+                            )
                         # Update user state
                         user.update_wallet(action_result)
                         self.update_analysis_dict()
                         self.run_trade_number += 1
-                        if self.verbose:
-                            print(
-                                "YieldSimulator.run_simulation:"
-                                f"\n\ttime = {self.market.time}"
-                                f"\n\tuser trade = {user_action}"
-                                f"\n\tuser_wallet = {user.wallet}"
-                                f"\n\tinit pool apy = {self.init_pool_apy}"
-                                f"\n\ttrades = {self.market.base_asset_orders + self.market.token_asset_orders}"
-                                f"\n\tinit_share_price = {self.market.init_share_price}"
-                                f"\n\tshare_price = {self.market.share_price}"
-                                f"\n\tamount = {self.trade_amount}"
-                                f"\n\treserves = {(self.market.share_reserves, self.market.bond_reserves)}"
-                            )
+                        last_user_action_time = self.market.time
+                        # if self.verbose:
+                            # print(
+                            #     "YieldSimulator.run_simulation:"
+                            #     f"\n\ttime = {self.market.time}"
+                            #     f"\n\tuser trade = {user_action}"
+                            #     f"\n\tuser_wallet = {user.wallet}"
+                            #     f"\n\tinit pool apy = {self.init_pool_apy}"
+                            #     f"\n\ttrades = {self.market.base_asset_orders + self.market.token_asset_orders}"
+                            #     f"\n\tinit_share_price = {self.market.init_share_price}"
+                            #     f"\n\tshare_price = {self.market.share_price}"
+                            #     f"\n\tamount = {self.trade_amount}"
+                            #     f"\n\treserves = {(self.market.share_reserves, self.market.bond_reserves)}"
+                            # )
+                if (self.market.time - last_user_action_time > 0.1) and (self.market.time - last_user_action_time) % 0.5 < 1/365/self.num_blocks_per_day/2:
```

Comment:
Shouldn't this be in a verbose conditional? If so, can you also include a TODO above that reminds us to fix this up when we improve logging.